src/python/pants/engine/build_files.py

```diff
@@ -252,29 +252,34 @@ def _hydrate(item_type, spec_path, **kwargs):
     )
 
   # NB: This may be empty, as the result of filtering by tag and exclude patterns!
-  return ProvenancedBuildFileAddresses(
-    ProvenancedBuildFileAddress(build_file_address=addr, provenance=addr_to_provenance[addr])
+  return AddressesWithOrigins(
+    AddressWithOrigin(address=addr, origin=addr_to_origin[addr])
     for addr in matched_addresses
   )
 
 
 @rule
-def remove_provenance(pbfas: ProvenancedBuildFileAddresses) -> BuildFileAddresses:
-  return BuildFileAddresses(pbfa.build_file_address for pbfa in pbfas)
+def remove_origins(addresses_with_origins: AddressesWithOrigins) -> BuildFileAddresses:
```

Comment:
```suggestion
def strip_origins(addresses_with_origins: AddressesWithOrigins) -> BuildFileAddresses:
```
`strip` is even a command-line tool, so this naming may also make more sense. Again, not super necessary though.

Reply:
I like the idea of `strip` as opposed to `remove`. What about `discard`?

My mental model of this all being a parser: at every stage, we decide to either "preserve" or "discard" information, such as preserving the original spec string vs. discarding that in favor of a normalized `Spec`. Here, we're saying that we no longer care about the origin, so we can discard that information.